include/librealsense2/dds/dds-device.cpp

```diff
@@ -344,46 +344,56 @@ size_t
 dds_device::foreach_video_profile( size_t sensor_index,
                                    std::function< void( const rs2_video_stream & profile, bool def_prof ) > fn ) const
 {
-    auto const & msg = _impl->_sensor_to_video_profiles.at( int( sensor_index ) );  // may throw!
-    for( size_t i = 0; i < msg.num_of_profiles; ++i )
+    auto const & msg = _impl->_sensor_to_video_profiles.find( int( sensor_index ) );
+    if(msg != _impl->_sensor_to_video_profiles.end())
     {
-        auto const & profile = msg.profiles[i];
-        rs2_video_stream prof;
-        prof.type   = profile.type;
-        prof.index  = profile.stream_index;
-        prof.uid    = profile.uid;
-        prof.width  = profile.width;
-        prof.height = profile.height;
-        prof.fps    = profile.framerate;
-        //TODO - bpp needed?
-        prof.fmt    = profile.format;
-        //TODO - add intrinsics
-        fn( prof, profile.default_profile );
+        for( size_t i = 0; i < msg->second.num_of_profiles; ++i )
+        {
+            auto const & profile = msg->second.profiles[i];
+            rs2_video_stream prof;
+            prof.type   = profile.type;
+            prof.index  = profile.stream_index;
+            prof.uid    = profile.uid;
+            prof.width  = profile.width;
+            prof.height = profile.height;
+            prof.fps    = profile.framerate;
+            //TODO - bpp needed?
+            prof.fmt    = profile.format;
+            //TODO - add intrinsics
+            fn( prof, profile.default_profile );
+        }
+
+        return msg->second.num_of_profiles;
     }
 
-    return msg.num_of_profiles;
+    return 0; //Not a video sensor, nothing to do
 }
 
 
 size_t
 dds_device::foreach_motion_profile( size_t sensor_index,
                                     std::function< void( const rs2_motion_stream & profile, bool def_prof ) > fn ) const
 {
-    auto const & msg = _impl->_sensor_to_motion_profiles.at( int( sensor_index ) );  // may throw!
-    for( size_t i = 0; i < msg.num_of_profiles; ++i )
+    auto const& msg = _impl->_sensor_to_motion_profiles.find( int( sensor_index ) );
```

Comment:
Why not return 0 if didn't find it and continue if all OK?

This will keep the alignment to the left and the handling will be done close to the condition?

Style comment, your choice if to leave it as is or not :)

Reply:
Done